webpack/components/Table/MainTable.js

```diff
@@ -13,24 +15,43 @@ import Loading from '../../components/Loading';
 const MainTable = ({
   status, cells, rows, error, emptyContentTitle, emptyContentBody,
   emptySearchTitle, emptySearchBody, searchIsActive, activeFilters,
-  ...extraTableProps
+  composable, rowsCount, children, ...extraTableProps
 }) => {
+  if (!composable && (!cells || !rows)) {
+    console.error(__('The <MainTable> component requires either a composable prop, or cells & rows props.')); // eslint-disable-line no-console
+  }
+
+  const tableHasNoRows = () => {
```

Comment:
I added a `rowsCount` prop to make up for the fact that we might not have a `rows` to check the length of. But I still check `rows.length` as before for non-composable tables, because that guarantees that the behavior won't have changed and that feels safer.